src/hotspot/share/utilities/globalDefinitions_gcc.hpp

```diff
@@ -31,12 +31,23 @@
 // globally used constants & types, class (forward)
 // declarations and a few frequently used utility functions.
 
+#if defined(_AIX)
+#include <alloca.h>
+#endif
 #include <ctype.h>
 #include <string.h>
 #include <stdarg.h>
 #include <stddef.h>
 #include <stdio.h>
 #include <stdlib.h>
+// In stdlib.h on AIX malloc is defined as a macro causing
+// compiler errors when resolving them in different depths as it
+// happens in the log tags. This avoids the macro.
+#if (defined(__VEC__) || defined(__AIXVEC)) && defined(AIX) \
+    && defined(__open_xl_version__) && __open_xl_version__ >= 17
+  #undef malloc
+  extern void *malloc(size_t) asm("vec_malloc");
+#endif
```

Comment:
This `#if` is not needed if we are building on AIX 7.2 TL5 SP7 or higher. This is the case in our build environment, but I think IBM is still building with SP5, which would run into build errors if I remove this `#if` now.

Reply:
Hi Thomas,
I would like to get totally rid of this, because as I mentioned IBM already modified the `stdlib.h` header not using `#define malloc vec_malloc` any more (and all the other vec_... defines). We have to ask the adoptium colleagues at IBM if they already have raised their build environment by the 2 SP levels needed.
In principle we had to do the same workaround for `calloc, free,...` too, but they didn't show up as errors in the logging files.
These lines where never meant to stay for long. Just to be able to compile until IBM fixes the issue, which is done now.

Reply:
@suchismith1993
Hi Suchi, can you please tell me when you will raise your build environment from AIX 7.2 TL5 SP5 to SP7?
I' am asking you, because I want to get rid of this nasty workaround.

Reply:
Pinging @sxa - what build environment does temurin use for AIX?

Reply:
Currently XLC16 but looking to upgrade to XLC17 on the minimum supported level for it (So it wouldn't be SP7 at present). Thanks for the ping - we have no current plans to increase to SP7.

Reply:
Seems like we need to keep it. This is unfortunate. I wouldn't risk mixing malloc and vec_malloc. Who knows what kind of problems this could cause?
What happens if we try to build this code on AIX 7.2 TL5 SP7? Will the compiler complain because `malloc` is no longer defined? Should we check `defined(malloc)` in addition?

Reply:
We already built this code since months on AIX 7.2 TL5 SP7, because we raised the OS.
This code is needed on SP5 and does not hurt SP7.